tests/model_registry/conftest.py

```diff
@@ -303,3 +316,57 @@ def generated_schema(model_registry_instance_rest_endpoint: str) -> Any:
         uri="https://raw.githubusercontent.com/kubeflow/model-registry/main/api/openapi/model-registry.yaml",
         base_url=f"https://{model_registry_instance_rest_endpoint}/",
     )
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_sessionstart(session: pytest.Session) -> None:
+    """Downloads external tests into the local project structure before test execution."""
+    print("pytest_sessionstart is running")
+    os.makedirs(UPSTREAM_TESTS_DIR, exist_ok=True)
+    for file in FILES:
+        os.system(f"curl -o {UPSTREAM_TESTS_DIR}/{file} {REPO_API_URL}/{FILE_PATH}/{file}")
+        print(f"Downloaded {file} into {UPSTREAM_TESTS_DIR}")
```

Comment:
This could be turned into a function that other people might use in the future if needed. I suggest moving it to a utility file for better reuse.

Reply:
This is a relatively specific hack to grab only a few specific files off of another repo, in theory you'd want to use gitpython to clone the repo and go from there. I can turn it into a function but I'm not really sure of the usefulness for other people

Reply:
My idea was to make it more modular, as it could be useful in the future for someone who might not be aware of it. While I can't guarantee that, I believe creating a function here would be beneficial. I do not have strong opinion this

Reply:
Done